index.d.ts

```diff
@@ -0,0 +1,44 @@
+declare module 'data-api-client' {
+    import type { ClientConfiguration } from 'aws-sdk/clients/rdsdataservice';
+
+    export interface iParams {
+        secretArn: string;
+        resourceArn: string;
+        database?: string;
+        keepAlive?: boolean;
+        hydrateColumnNames?: boolean;
+        sslEnabled?: boolean;
+        options?: ClientConfiguration;
+        region?: string;
+    }
+
+    interface iTransaction {
+        query: iQuery<iTransaction>;
+        rollback: (error: Error, status: any) => void;
+        commit: () => Promise<void>;
+    }
+
+    interface iQuery<T> {
+        (sql: string, params?: [] | unknown): Promise<T>; // params can be [] or {}
```

Comment:
I think for the generics, you can do something like this (instead of making `iQuery` itself a generic):
```suggestion
        <T = any>(sql: string, params?: [] | unknown): Promise<T>; // params can be [] or {}
```
Usage:
```ts
client.query<{ name: string; age: number }[]>('SELECT * from table';)
```

Reply:
Just realised this is the same as @rraziel 's suggestion 😅